mcs/class/referencesource/System/services/monitoring/system/diagnosticts/Process.cs

```diff
@@ -230,7 +230,10 @@ private enum StreamReadMode
             get {
                 EnsureState(State.Exited);
 #if MONO
-                if (exitCode == -1)
+		
+		int platform = (int)Environment.OSVersion.Platform;
+		
+                if (( (platform == 4) || (platform == 6) || (platform == 128) ) && exitCode == -1)
```

Comment:
Can we copy the OnUnix or whatever function from somewhere?

Reply:
Yes, ofcource, I'll do it later. And I'm thinking about full deletion of this part of code because many programs use exit code -1...

P.S. Sorry for bad English!

Reply:
Just do `if (exitCode == -1 && !RuntimeInformation.IsOSPlatform (OSPlatform.Windows))`